mobify/documentation-theme
# Styles For Upgraded Styleguidist

With Styleguidist upgraded to the latest (and ditching our fork), some new styles had to be written to get it to work in our docs!

 **Tickets**: https://mobify.atlassian.net/browse/WEB-1552
 **PRs**: [progressive-web-sdk#1029](https://github.com/mobify/progressive-web-sdk/pull/1029)

## Changes

- Wrapping `.c-contain.c-text-content` in a `u-context`, which allows us to position Styleguidist's sidebar navigation the way we want
- New and modified styles in `_styleguidist.scss`

| Before | After |
| --- | --- |
| ![screen shot 2018-01-31 at 5 16 55 pm](https://user-images.githubusercontent.com/734535/35656161-9c7d7e54-06aa-11e8-8a83-f394c68bbc3a.png) | ![screen shot 2018-01-31 at 5 16 58 pm](https://user-images.githubusercontent.com/734535/35656167-a08d7a8a-06aa-11e8-9224-47dbadec984f.png) |

## How To Test

1. Pull down this branch (`upgrade-styleguidist`)
2. Change to this repo's directory
3. Run `npm link` or `yarn link`
4. Change to the Progressive Web SDK repo
5. Pull down the PR's branch (`styleguidist-upgrade`, note: not the same as this repo's branch)
6. Run `rm -rf node_modules`
7. Run `npm install` or `yarn install`
8. Run `npm link @mobify/documentation-theme` or `yarn link @mobify/documentation-theme`
9. Run `npm run docs:dev`
10. Visit [http://localhost:9000/latest/components/](http://localhost:9000/latest/components/)
11. **Verify** that the design works as expected
